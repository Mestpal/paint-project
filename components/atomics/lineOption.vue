<template>
  <div
    @click="lineWeightOptionSelected"
  >
    LINE {{ lineConf.weight }}
  </div>
</template>

<script>
export default {
  props: {
    weight: {
      type: Number,
      default: 0
    }
  },
  data () {
    return {
      lineConf: {
        weight: this.weight
      }
    }
  },
  methods: {
    lineWeightOptionSelected () {
      this.$emit('lineEvent', this.lineConf)
    }
  }
}
</script>
