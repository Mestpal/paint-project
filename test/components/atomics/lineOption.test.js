
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'

import lineOption from '@/components/atomics/lineOption'

Vue.use(Vuetify)
Vue.use(Vuex)
const localVue = createLocalVue()

describe('lineOption.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(lineOption, {
      localVue,
      propsData: {
        weight: 128
      }
    })
  })

  it('reders the lineOption component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit an event when the line option is clicked', () => {
    const event = jest.fn()
    const lineOptionDiv = wrapper.find('div')

    wrapper.vm.$on('lineEvent', event)

    expect(event).toHaveBeenCalledTimes(0)

    // Simulate a click on the color
    lineOptionDiv.trigger('click')

    // Ensure that our mock event was called
    expect(event).toHaveBeenCalledTimes(1)
  })
})
