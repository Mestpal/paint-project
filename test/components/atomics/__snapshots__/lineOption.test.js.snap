// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lineOption.vue reders the lineOption component correctly 1`] = `
<div>
  LINE 128
</div>
`;
